includes/class-wc-google-gtag-js.php

```diff
@@ -49,10 +49,40 @@ public function __construct( $settings = array() ) {
 		$this->map_hooks();
 
 		// Setup frontend scripts
+		add_action( 'wp_head', array( $this, 'setup_site_tag' ), 2 );
 		add_action( 'wp_enqueue_scripts', array( $this, 'enquque_tracker' ), 5 );
 		add_action( 'wp_footer', array( $this, 'inline_script_data' ) );
 	}
 
+	/**
+	 * Setup the global site tag as early as possible on the page
+	 *
+	 * @return void
+	 */
+	public function setup_site_tag() {
+		// phpcs:disable WordPress.WP.EnqueuedResources.NonEnqueuedScript
+		printf(
```

Comment:
I wonder if we should be so brutal to break WP rules and print script directly; maybe enqueueing it as an inline with/ no dependencies would be enough? Or does the "optimization" plugins could force scripts enqueued in the head to be loaded later?

Reply:
Some of those plugins have settings to combine all inline JS and some will move them to a different file. As we know the script needs to be available at this point no matter what I think it makes sense that we avoid that possibility.

Instead of outputting it directly I've wrapped it in `wp_print_inline_script_tag` so that's we're still doing things the WordPress way 582e41b